src/client.ts

```diff
@@ -42,34 +42,80 @@ export class OpenFeatureClient implements Client {
     this._context = context;
   }
 
-  set logger(logger: Logger) {
+  /**
+   * Sets a logger on the client. This logger supersedes to the global logger
+   * and is passed to various components in the SDK.
+   *
+   * @param {Logger} logger The logger to to be used
+   * @returns {OpenFeatureClient} OpenFeature Client
+   */
+  setLogger(logger: Logger): OpenFeatureClient {
     this._clientLogger = new SafeLogger(logger);
+    return this;
   }
 
-  get logger(): Logger {
-    return this._clientLogger || this.globalLogger();
-  }
-
-  set context(context: EvaluationContext) {
+  /**
+   * Sets client evaluation context that will be used during flag evaluations
```

Comment:
```suggestion
   * Sets the evaluation context that will be used during flag evaluations
```

Reply:
I think it' important to highlight that this **IS** scoped to the client. Perhaps it could be rephased to something like:
`Settings the evaluation context on the client so it can be used during flag evaluation.`

Reply:
Sets the evaluation context that will be used during flag evaluations on this client.